snuba/util.py

```diff
@@ -183,17 +187,27 @@ def complex_column_expr(expr, body, depth=0):
     out = []
     i = 0
     while i < len(args):
-        next_2 = args[i:i+2]
-        if is_function(next_2, depth+1):
-            out.append(complex_column_expr(next_2, body, depth+1))
+        next_2 = args[i:i + 2]
+        if is_function(next_2, depth + 1):
+            out.append(complex_column_expr(next_2, body, depth + 1))
             i += 2
         else:
             nxt = args[i]
-            assert not isinstance(nxt, (list, tuple)), "nested functions do not support array arguments"
-            if isinstance(nxt, six.string_types):
-                out.append(column_expr(nxt, body))
+            # TODO(LB): Added this because is_condition throws an error if a literal is passed,
+            # should it be a safe in that case?
+            if isinstance(nxt, (list, tuple)) and is_condition(nxt):
+                out.append(conditions_expr([nxt], body, depth=0))
             else:
-                 out.append(escape_literal(nxt))
+                if isinstance(nxt, (list, tuple)):
+                    next_2 = nxt[0:2]
```

Comment:
This new block seems duplicated from the logic above? why isn't this just handled by the recursive call to `column_expr()` (which will call `complex_column_expr()` again if needed)?